client/src/pages/index.tsx

```diff
@@ -35,32 +36,50 @@ const Homepage: NextPage<FeedsPageInitials> = (props): JSX.Element => {
 	const [isLoading, setIsLoading] = useState(true)
 	const [isLoadMoreTriggered, setIsLoadMoreTriggered] = useState(false)
 
-	const handleCategoryFilterSet = (id) => {
+	useEffect(() => {
+		handleDataFetching().then(() => setIsJustInitialized(false))
+	}, [])
+
+	useEffect(() => {
+		if (!isJustInitialized) {
+			handleDataFetching()
+		}
+	}, [skipValueForPagination, tagFilter, sortBy, isLoading])
+
+	const handleFlowReset = (): void => {
 		setIsLoading(true)
 		setFeed([])
 		setSkipValueForPagination(0)
-		setIsJustInitialized(false)
-		setCategoryFilter(id)
 	}
 
-	const handleSortBySet = (val) => {
-		setIsLoading(true)
-		setFeed([])
-		setSkipValueForPagination(0)
-		setIsJustInitialized(false)
-		setSortBy(val)
+	const handleSorting = (sortBy: 'hot' | 'top' | undefined): void => {
+		handleFlowReset()
+		setSortBy(sortBy)
+	}
+
+	const handleFilteringTags = (tag: string): void => {
+		if (tagFilter) {
```

Comment:
How about this?

```		
let tagFilter = tag

// This could also use a custom method with clearer context naming. 
if (tagFilter && tagFilter.split(',').includes(tag) === false) {
     tagFilter = `${tagFilter},${tag}`
}

handleFlowReset()
setTagFilter(tag)
```